src/rgw/rgw_asio_frontend.cc

```diff
@@ -210,16 +211,22 @@ void handle_connection(boost::asio::io_context& context,
       RGWRestfulIO client(cct, &real_client_io);
       auto y = optional_yield{context, yield};
       int http_ret = 0;
+      const auto started = ceph::coarse_real_clock::now();
+
+      auto time_t_started = ceph::coarse_real_clock::to_time_t(started);
+      const tm* local_tm = localtime(&time_t_started);
+      char apache_formatted_time [80];
+      strftime(apache_formatted_time,80,"%d/%b/%Y:%T %z",local_tm);
+
       process_request(env.store, env.rest, &req, env.uri_prefix,
                       *env.auth_registry, &client, env.olog, y,
                       scheduler, &http_ret);
 
       if (cct->_conf->subsys.should_gather(dout_subsys, 1)) {
         // access log line elements begin per Apache Combined Log Format with additions following
-        const auto now = ceph::coarse_real_clock::now();
         using ceph::operator<<; // for coarse_real_time
         ldout(cct, 1) << "beast: " << hex << &req << dec << ": "
-            << remote_endpoint.address() << " - - [" << now << "] \""
+            << remote_endpoint.address() << " - - [" << apache_formatted_time << "] \""
```

Comment:
Looks good, but we need to add sub-second accuracy, possibly like that:
```suggestion
            << remote_endpoint.address() << " - - [" << put_time(local_tm, "%d/%b/%Y:%T.")
            << setfill('0') << setw(3) << std::chrono::duration_cast<std::chrono::milliseconds>(started.time_since_epoch()).count() % 1000
            << put_time(local_tm, " %z") << "] \""
```

Reply:
updated awk parser:
```
❯ tail -n 25 ./out/radosgw.8000.log | grep "beast:" | awk -F '[ \\[\\]]' '{ cmd_date_done = "date +%s.%N -d " $1 ; gsub(/\//,"-",$11); sub(/:/," ",$11); cmd_date_start = "date +%s.%N -d \"" $11$12 "\"" ; DONETIME=0 ; cmd_date_done | getline DONETIME ; close(cmd_date_done) ; STARTTIME=0 ; cmd_date_start | getline STARTTIME ; close(cmd_date_start) ; LATENCY = DONETIME - STARTTIME ; print "done: " $1 " - start: " $11$12 " = latency: " LATENCY "s" }'
done: 2020-11-04T12:47:56.334+0200 - start: 04-Nov-2020 12:47:56.322+0200 = latency: 0.0120001s
done: 2020-11-04T12:47:56.335+0200 - start: 04-Nov-2020 12:47:56.324+0200 = latency: 0.0110002s
done: 2020-11-04T12:47:56.337+0200 - start: 04-Nov-2020 12:47:56.322+0200 = latency: 0.0149999s
done: 2020-11-04T12:47:56.338+0200 - start: 04-Nov-2020 12:47:56.325+0200 = latency: 0.013s
done: 2020-11-04T12:47:56.340+0200 - start: 04-Nov-2020 12:47:56.327+0200 = latency: 0.013s
done: 2020-11-04T12:47:56.340+0200 - start: 04-Nov-2020 12:47:56.329+0200 = latency: 0.0109999s
done: 2020-11-04T12:47:56.341+0200 - start: 04-Nov-2020 12:47:56.330+0200 = latency: 0.0110002s
done: 2020-11-04T12:47:56.345+0200 - start: 04-Nov-2020 12:47:56.334+0200 = latency: 0.0109999s
done: 2020-11-04T12:47:56.346+0200 - start: 04-Nov-2020 12:47:56.330+0200 = latency: 0.016s
done: 2020-11-04T12:47:56.348+0200 - start: 04-Nov-2020 12:47:56.336+0200 = latency: 0.0120001s
done: 2020-11-04T12:50:12.938+0200 - start: 04-Nov-2020 12:50:12.935+0200 = latency: 0.00300002s
❯ tail -n 25 ./out/radosgw.8000.log | grep "latency="
2020-11-04T12:47:56.334+0200 7fa56d670700  1 ====== req done req=0x7fa4e735fa70 op status=0 http_status=200 latency=0.011999929s ======
2020-11-04T12:47:56.335+0200 7fa56b66c700  1 ====== req done req=0x7fa4e7563a70 op status=0 http_status=200 latency=0.010999935s ======
2020-11-04T12:47:56.337+0200 7fa57be8d700  1 ====== req done req=0x7fa4e74e2a70 op status=0 http_status=200 latency=0.013999917s ======
2020-11-04T12:47:56.338+0200 7fa54fe35700  1 ====== req done req=0x7fa4e73e0a70 op status=0 http_status=200 latency=0.012999923s ======
2020-11-04T12:47:56.340+0200 7fa575e81700  1 ====== req done req=0x7fa4e725da70 op status=0 http_status=200 latency=0.012999923s ======
2020-11-04T12:47:56.340+0200 7fa565660700  1 ====== req done req=0x7fa4e7461a70 op status=0 http_status=200 latency=0.010999934s ======
2020-11-04T12:47:56.341+0200 7fa556e43700  1 ====== req done req=0x7fa4e71dca70 op status=0 http_status=200 latency=0.010999935s ======
2020-11-04T12:47:56.345+0200 7fa569e69700  1 ====== req done req=0x7fa4e735fa70 op status=0 http_status=200 latency=0.010999935s ======
2020-11-04T12:47:56.346+0200 7fa540e17700  1 ====== req done req=0x7fa4e72dea70 op status=0 http_status=200 latency=0.015999905s ======
2020-11-04T12:47:56.348+0200 7fa55d650700  1 ====== req done req=0x7fa4e7563a70 op status=0 http_status=200 latency=0.011999929s ======
2020-11-04T12:50:12.938+0200 7fa534dff700  1 ====== req done req=0x7fa4e7563a70 op status=0 http_status=200 latency=0.002999982s ======
```